src/js/headerMiddleware.js

```diff
@@ -0,0 +1,69 @@
+/*
+
+    A `gpii.express.middleware` component that sets one or more HTTP headers based on the content of `options.headers`.
+
+    See the documentation for more details:
+
+    https://github.com/GPII/gpii-express/blob/master/docs/headerMiddleware.md
+
+ */
+"use strict";
+var fluid = require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+
+fluid.registerNamespace("gpii.express.middleware.headerSetter");
+
+/**
+ *
+ * @param that {Object} The middleware component itself.
+ * @param req {Object} The Express request object.
+ * @param res {Object} The Express response object.
+ * @param next {Function} The function to be executed next in the middleware chain.
+ *
+ * A middleware function that adds one or more headers before launching the next middleware function in the chain.
+ *
+ */
+gpii.express.middleware.headerSetter.addHeaders = function (that, err, req, res, next) {
+    fluid.each(that.options.headers, function (headerOptions) {
+        var templateData = fluid.model.transformWithRules({ that: that, request: req }, headerOptions.dataRules);
+        var fieldValue = fluid.stringTemplate(headerOptions.template, templateData);
+
+        res.setHeader(headerOptions.fieldName, fieldValue);
+    });
+
+    if (err) {
```

Comment:
Horrid, isn't it : P
You may like, for example. next.apply(null, fluid.makeArray(err)) but you may reject it as being "cutesy"